src/main/java/pl/psnc/dei/ui/components/batches/ComplexBatchImportComponent.java

```diff
@@ -183,10 +178,17 @@ private void prepareImportName() {
         }
     }
 
+    private void removeOldImportName() {
+        if (ImportNameCreatorUtil.isMatchingImportTitlePattern(this.nameTextFiled.getValue())) {
+            this.nameTextFiled.setValue("");
+        }
+    }
+
     private void sendImport() {
         try {
-            this.prepareImportName();
             if (this.validate()) {
+                this.removeOldImportName();
+                this.prepareImportName();
```

Comment:
I still highly recommend to not put this all _import name creation_ logic to the user interface component. In my opinion, the easiest and safest solution is just to pass the import name to the `BatchService#makeComplexImport` and make it deal with the empty name of import by creating new, following the pattern.
In the end, you receive list of imports with generated or passed import name. Then you can display it to the user in notification. I think this way is much better, because you can display names of all of the created imports.

Reply:
I agree, logic should be in the service.